src/createAgentManager.ts

```diff
@@ -106,136 +132,161 @@ export function getAgent(agentId: string, auth: Auth, baseURL?: string): Promise
-        },
-        disconnect() {
-            analytics.track('agent-chat', {
-                event: 'disconnect',
-                chatId: chat.id,
-                agentId: typeof agent === 'string' ? agent : agent.id,
-            });
+            analytics.track('agent-chat', { event: 'connect', chatId: chat.id, agentId: agentInstance.id });
 
-            abortController.abort();
-            socketManager.disconnect();
+            items.streamingManager = streamingManager;
+            items.socketManager = socketManager;
+            items.chat = chat;
+            items.messages = getInitialMessages(agentInstance);
 
-            return streamingManager.disconnect();
+            options.callbacks.onNewMessage?.(items.messages);
         },
+        async disconnect() {
+            items.socketManager?.disconnect();
+            await items.streamingManager?.disconnect();
 
-        chat(messages: Message[], append_chat: boolean = false) {
-            if (messages.length === 0) {
-                throw new Error('Messages cannot be empty');
+            analytics.track('agent-chat', { event: 'disconnect', chatId: items.chat?.id, agentId: agentInstance.id });
+        },
+        async chat(userMessage: string, append_chat: boolean = false) {
+            if (userMessage.length === 0) {
+                throw new Error('Message cannot be empty');
             }
-            analytics.track('agent-message-send', {
-                event: 'success',
-                messages: messages.length + 1,
-            });
 
-            messages[messages.length - 1].created_at = new Date().toISOString();
-            if (!messages[messages.length - 1].role) {
-                messages[messages.length - 1].role = 'user';
+            if (!items.chat) {
+                throw new Error('Chat is not initialized');
             }
 
-            return agentsApi.chat(
-                agentInstance.id,
-                chat.id,
-                { sessionId: streamingManager.sessionId, streamId: streamingManager.streamId, messages, append_chat },
-                { signal: abortController.signal }
-            );
+            if (!items.streamingManager) {
+                throw new Error('Streaming manager is not initialized');
+            }
+
+            analytics.track('agent-message-send', { event: 'success', messages: items.messages.length + 1 });
+            items.messages.push({
+                id: getRandom(),
+                role: 'user',
+                content: userMessage,
+                created_at: new Date().toISOString(),
+            });
+
+            const response = await agentsApi.chat(agentInstance.id, items.chat.id, {
+                sessionId: items.streamingManager.sessionId,
+                streamId: items.streamingManager.streamId,
+                messages: items.messages,
+                append_chat,
+            });
+
+            items.messages.push({
+                id: getRandom(),
```

Comment:
if user dont send it throw the API?
is the API create ID for message in BE - if not please add